app/jinja_filters.py

```diff
@@ -43,6 +43,18 @@ def format_date(value):
     return value.strftime('%-d %B %Y')
 
 
+@blueprint.app_template_filter()
+def format_conditional_date(dates):
+    """ This function format_cond_date accepts two dates, if a date is submitted that is used and is available,
+    if not the metadata date is made available The format (example) in the json is
+    {{[answers.date_from, exercise.start_date]|format_cond_date}} """
+
+    if dates[0]:
+        return format_str_as_short_date(dates[0])
+    else:
+        return format_date(dates[1])
```

Comment:
Dates in answers are stored as a string in a `dd/mm/yyyy` format, whereas dates in the metadata and survey contexts are stored as Python date objects. 

As this method should be able to compare any dates, it should deal with both date formats in either argument. I think for each date it would be fine to check if it's a Python date object and if not assume it's a date in a `dd/mm/yyyy` format.

This has highlighted some inconsistencies in the way that we handle dates that we should review.